Fix #21: Add setup files to create Python package

nvs-abhilash opened the pull request:
Added setup.py file, and MANIFEST.in file to create a .whl file.
This helps in making the CorrectMe as a package.
Run `python setup.py sdist bdist_wheel` to get the wheel in `dist` directory.
This fixed #21.

piyush-jaiswal commented on README.md:
```diff
@@ -11,21 +11,24 @@ All contributions are welcomed :)
+conda create -n cm_env python=3.6 pip
+source activate cm_env # activate cm_env for windows
+cd CorrectMe # The cloned repository directory
+python -m pip install --user --upgrade setuptools wheel
```
I can't figure out why we need to build wheels manually. Couldn't we simply use these commands instead - ```pip install <path_to_cloned_repo>``` or ```python setup.py install```?

nvs-abhilash replied:
Yes, you are right! My fault, did not do much digging on `setup.py`. Updated the README, though I was not able to install using git directly. I guess I am having some syntax issues. Do you know any way to do it directly from git?

piyush-jaiswal replied:
I think this should work ```pip install git+https://github.com/nvs-abhilash/CorrectMe.git```.

nvs-abhilash replied:
It doesn't seem to be working for me. Getting this error:
![error_correctme](https://user-images.githubusercontent.com/15072945/49693664-4a511600-fb9f-11e8-8b0a-cac2151b6bd6.png)

This seems to be a issue with PIP: [https://github.com/pypa/pip/issues/5271](https://github.com/pypa/pip/issues/5271) 

Have you been facing similar issue or is it working for you?

piyush-jaiswal replied:
The git URL is for the `master` branch, which doesn't have `setup.py` yet, hence the error. You have to specify the branch in the URL if its not master. Try this `pip install git+https://github.com/nvs-abhilash/CorrectMe.git@installation`.

piyush-jaiswal commented:
Since we only support python3, we could also use the ```python_requires``` argument of ```setup``` from ```setuptools```, namely ```python_requires='>=3'```

piyush-jaiswal commented:
Should we include ```main.py``` in the package as well?

nvs-abhilash commented:
Not sure how to add `main.py` to the package, as an example script or something? Should we rename it to something more meaningful inside package? We also would need a way to add the data folder to the package as well for `main.py`. Do you have some suggestions?

nvs-abhilash commented on setup.py:
```diff
@@ -22,7 +22,7 @@
     url='https://github.com/nvs-abhilash/CorrectMe',  # Optional
 
-    author='Piyush Jaiswal and NVS Abhilash',  # Optional
+    author='NVS Abhilash and Piyush Jaiswal',  # Optional
```
-_-

piyush-jaiswal commented:
Renamed `main.py` to `Correctme.py` and included in the installation. Included `data` folder to be part of the package. @nvs-abhilash, let me know if we need more changes.

piyush-jaiswal commented:
Hey @nvs-abhilash, LGTM. Should I merge this?

nvs-abhilash commented:
Sorry did not get a chance to test the installation properly. Will do it by tomorrow, and then let's merge it.

piyush-jaiswal commented:
Yeah sure. Take your time. I just thought you were waiting on me to merge this :)

nvs-abhilash commented:
Hi @piyush-jaiswal, finally got chance to look at it. Though the installation was working fine, but I was still confused on how to call `CorrectMe.py` and when I examined the installation I found out that it was getting installed outside the package, not sure whether it was necessary.

Then after some research, I stumbled across this article: [python-apps-the-right-way-entry_points-and-scripts](https://chriswarrick.com/blog/2014/09/15/python-apps-the-right-way-entry_points-and-scripts/).

The above article showed how we can use python scripts as executable from the command prompt. And I guess this would allow us to call something like `correctme-gui` from the terminal itself and it would work as before!! I think this is awesome and exactly what we needed :1st_place_medal: . I have updated the `setup.py` file and moved `CorrectMe.py` to `bin` folder inside the package.

I request you to check if it is working for you or not, and suggest further improvements to the way I wrote it (maybe some naming choices), then let's close this.
